### PR TITLE
Don't fail an incremental build if can't find a merge base

### PR DESCRIPTION
**Pull request by fkorotkov**
Added an additional check and a warning message.

This change addresses an issue experienced in #423.

/cc @goderbauer

### CODE REVIEW

File: script/incremental_build.sh [hunk 1]
```diff
@@ -10,10 +10,20 @@ if [ "${BRANCH_NAME}" = "master" ]; then
 else
   # Make sure there is up-to-date master.
   git fetch origin master
-  BRANCH_BASE_SHA=$(git merge-base --fork-point FETCH_HEAD HEAD)
-  echo "Checking changes from $BRANCH_BASE_SHA..."
-  FLUTTER_CHANGED_GLOBAL=`git diff --name-only $BRANCH_BASE_SHA HEAD | grep -v packages | wc -l`
-  FLUTTER_CHANGED_PACKAGES=`git diff --name-only $BRANCH_BASE_SHA HEAD | grep -o "packages/[^/]*" | sed -e "s/packages\///g" | sort | uniq | paste -s -d, -`
+
+  FLUTTER_CHANGED_GLOBAL=0
+  FLUTTER_CHANGED_GLOBAL=""
```
Comment (goderbauer) [inline script/incremental_build.sh]: Is it intentional to first assign 0 and then empty string?

Comment (fkorotkov) [reply script/incremental_build.sh]: Good catch! It's a typo. It suppose to be `FLUTTER_CHANGED_PACKAGES`. Updated the PR.

File: script/incremental_build.sh [hunk 2]
```diff
@@ -10,10 +10,20 @@ if [ "${BRANCH_NAME}" = "master" ]; then
...
+  FLUTTER_CHANGED_PACKAGES=""
+
+  # Try get a merge base for the branch and calculate affected packages.
+  # We need this check because some CIs can do a single branch clones with a limited history of commits.
+  if BRANCH_BASE_SHA=$(git merge-base --fork-point FETCH_HEAD HEAD); then
+    echo "Checking changes from $BRANCH_BASE_SHA..."
+    FLUTTER_CHANGED_GLOBAL=`git diff --name-only $BRANCH_BASE_SHA HEAD | grep -v packages | wc -l`
+    FLUTTER_CHANGED_PACKAGES=`git diff --name-only $BRANCH_BASE_SHA HEAD | grep -o "packages/[^/]*" | sed -e "s/packages\///g" | sort | uniq | paste -s -d, -`
+  else
+    echo "Cannot find a merge base for the current branch to run an incremental build!"
```
Comment (goderbauer) [inline script/incremental_build.sh]: Can you add to this message what would be required to make the build incremental (e.g. rebase, maybe?)?

**Comment by fkorotkov** [comment]
@goderbauer added an actionable message 🙏

**Comment by goderbauer** [comment]
LGTM

Travis failed, I assume its a flake. I restarted it, will merge on green.

**Pull request merged**